### DISCUSSION TITLE
Waveshare Display w/Nordic-Zephyr, spi

### DISCUSSION THREAD

**Discussion by mej558** [zephyrproject-rtos/zephyr#73738] [Q&A]
I would really appreciate help getting a waveshare display to work with a nordic nrf52840dk and zephyr. We were able to get the display working within minutes in Arduino, but it's been weeks struggling with nordic/zephyr.

https://github.com/mej558/lvgl_1.git

- 2.13inch E-Ink Display HAT V4 Version, Black/White Two-Color 250x122 (from amazon)
- There are no recognizable controller marking on the FPC cable. Ours only reads: FPC-A002
- So, we think we should use this DSHIELD: west build -b nrf52840dk_nrf52840 -d build52840  -- -DSHIELD=waveshare_epaper_gdeh0213b72
- https://docs.zephyrproject.org/latest/boards/shields/waveshare_epaper/doc/index.html
- We are using SPI3 pinouts +those designated in the Waveshare DSHIELD, two people have set it up in dependently and correct pins has been triple-checked (as far as we understand the connections)
- Using the LVGL sample
- We get no response at all from the screen
- I noticed that zephyr_display is not defined: display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
- only ~6months into Nordic/Zephyr so we get the basics, but still learning stuff

Any help or guidance would be appreciated. If I have left out anything important please let me know and I will add it.

![Zephyr_Display](https://github.com/zephyrproject-rtos/zephyr/assets/80479572/c554ba61-8b76-4495-89f9-52650b19a5e6)
![IMG_0731](https://github.com/zephyrproject-rtos/zephyr/assets/80479572/f9cd67ea-e5ea-43dd-b685-ca436f342ca1)
![nrf52840dk-waveshare spi3 connections](https://github.com/zephyrproject-rtos/zephyr/assets/80479572/68e6b551-66d1-4c82-b54a-3e4ecc291cbf)

**Comment by WouterHielkema** [comment]
Hey! I'm doing a project with a pervasive display and the nrf52840dk now! Have you ever figured this out? Any more information you can share with me to help me along? I'd love to hear from you! 

-Wouter

**Comment by mej558** [comment]
If I remember correctly, the connections as shown are correct. 
I think our problem was understanding that you need to compile in the proper shield for your display. And, that there were a limited number of Zephyr supported Waveshare displays?
This folder was important to understanding: C:\ncs\v2.9.1\zephyr\boards\shields\waveshare_epaper
Depending on the display(shield), the command to build is something similar to:  
west build -b nrf52840dk_nrf52840 -d build_nrf52840dk_nrf52840 -p -- -DSHIELD=waveshare_epaper_gdey0213b74

Its been a while, so I am a little fuzzy on the details, but I hope that helps. Good luck with your project!
